Classes/EventListener/NewsLayoutListener.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+declare(strict_types=1);
+
+namespace B13\Newspage\EventListener;
+
+/*
+ * This file is part of TYPO3 CMS-based extension "newspage" by b13.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ */
+
+use TYPO3\CMS\Backend\Controller\Event\ModifyPageLayoutContentEvent;
+use TYPO3\CMS\Backend\Form\FormDataCompiler;
+use TYPO3\CMS\Backend\Form\FormDataGroup\TcaDatabaseRecord;
+use TYPO3\CMS\Backend\Form\FormResultCompiler;
+use TYPO3\CMS\Backend\Form\NodeFactory;
+use TYPO3\CMS\Backend\Routing\UriBuilder;
+use TYPO3\CMS\Backend\Template\Components\ButtonBar;
+use TYPO3\CMS\Backend\Template\ModuleTemplate;
+use TYPO3\CMS\Backend\Utility\BackendUtility;
+use TYPO3\CMS\Core\Authentication\BackendUserAuthentication;
+use TYPO3\CMS\Core\Configuration\ExtensionConfiguration;
+use TYPO3\CMS\Core\Database\Connection;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Database\Query\Restriction\DeletedRestriction;
+use TYPO3\CMS\Core\Database\Query\Restriction\WorkspaceRestriction;
+use TYPO3\CMS\Core\Imaging\Icon;
+use TYPO3\CMS\Core\Imaging\IconFactory;
+use TYPO3\CMS\Core\Localization\LanguageService;
+use TYPO3\CMS\Core\Page\PageRenderer;
+use TYPO3\CMS\Core\Type\Bitmask\Permission;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+final class NewsLayoutListener
+{
+    public function __construct(
+        protected readonly PageRenderer $pageRenderer,
+        protected readonly IconFactory $iconFactory,
+        protected readonly ExtensionConfiguration $extensionConfiguration
+    ) {}
+
+    public function __invoke(ModifyPageLayoutContentEvent $event): void
+    {
+        if(!$this->extensionConfiguration->get('newspage', 'layout_edit_mode')) {
+            return;
+        }
+
+        $queryParams = $event->getRequest()->getQueryParams();
+        $pageId = (int)($queryParams['id'] ?? 0);
+        $language = (int)($queryParams['language'] ?? 0);
+        $function = (int)($queryParams['function'] ?? 1);
+        $pageInfo = BackendUtility::readPageAccess($pageId, $this->getBackendUser()->getPagePermsClause(Permission::PAGE_SHOW));
+
+        // Display page property inline edit only for doktype=24 and function=1 (layout mode)
+        if($function !== 1 || $pageInfo['doktype'] !== 24 || !$this->isPageEditable($language, $pageInfo)) {
```

Comment:
Could we use a constant for the doktype?

Reply:
Done.